project/BuildPlugin.scala

```diff
@@ -79,25 +82,30 @@ object BuildKeys {
     Keys.test in AssemblyKeys.assembly := {}
   )
 
+  import sbtbuildinfo.BuildInfoKeys
   val benchmarksSettings: Seq[Def.Setting[_]] = List(
     Keys.skip in Keys.publish := true,
+    BuildInfoKeys.buildInfoKeys := Seq[BuildInfoKey](
```

Comment:
If you need a reference to the project from here and `javaOptions`, I would add a method in `BuildDefaults` that takes a project as a parameter. Then, I would use that reference to scope whichever settings you need and return a list of settings that you apply from `build.sbt`. I think this is way better than trusting that we use a variable for id and for `LocalProject`, and I think it's also more extensible :smile: